contracts/erc721/PaymentObligation.sol

```diff
@@ -39,23 +39,24 @@ contract PaymentObligation is Initializable, UserMintableERC721 {
   external
   view
   returns (
-    string memory grossAmount,
-    string memory currency,
-    string memory dueDate,
+    bytes memory grossAmount,
+    bytes memory currency,
+    bytes memory dueDate,
     uint256 anchorId,
     bytes32 documentRoot
   )
   {
     anchorId = tokenDetails_[_tokenId].anchorId;
     return (
-    poDetails_[anchorId].grossAmount,
-    poDetails_[anchorId].currency,
-    poDetails_[anchorId].dueDate,
-    anchorId,
-    tokenDetails_[_tokenId].rootHash
+      poDetails_[anchorId].grossAmount,
```

Comment:
why are the podetails identified by anchorId and not tokenId?

Reply:
This comment is in the code:
```
 use the anchorId as a key in order to prevent double minting for the same anchor
```

If i think about it we do not need this anymore as we have the one nft per document document proof.

Reply:
I think that's better. Especially because we want to allow some contracts to mint multiple per document if they implement it.